### ISSUE TITLE
no-unused-variable function called from the view Angular 2.

### ISSUE THREAD

**Issue by Kavorka79** [palantir/tslint#1295]
### Bug Report? Question

- __TSLint version__: `3.10.2`
- __TypeScript version__: `1.8.10`
- __Running TSLint via__: gulp task

#### TypeScript code being linted

```ts

@Component({
    templateUrl: /someComponent.html",
})
export class SomeComponent {
    private cancel(): void {
        console.log("CANCEL!");
    }
}
```
```html
<button type="button" (click)="cancel()" >Avbryt</button>
```

with `tslint.json` configuration:

```json
    "no-unused-variable": [
      true,
      "check-parameters",
      { "ignore-pattern": "fdescribe|xdescribe|fit|xit|on*|/g" }
    ],
```

#### Actual behavior

reported as an unused variable

#### Expected behavior

Could it check specified view file?
solved by prefixin with on but seems like there should be some better solution but have no Idea what....

**Comment by adidahiya** [comment]
Checking a template file for references feels out-of-scope for TSLint because we rely on TypeScript's Language Service APIs for finding occurrences of values. I don't think TSLint will do this unless the LS does static analysis of templates (there is _some_ hope for this; see https://github.com/Microsoft/TypeScript/issues/3003 and https://github.com/Microsoft/TypeScript/issues/5151).

In the meantime, your workarounds are to use the `ignore-pattern` option or make those methods `public`.

**Comment by Kavorka79** [comment]
Thats what I suspected, but wanted to make sure. Also relates to [Microsoft/Typescript#5470 ](https://github.com/Microsoft/TypeScript/issues/5470).

Thanks!

**Comment by JKillian** [comment]
Going to close this because it's not actionable on the TSLint side of things. As always, more discussion is always welcome and if thing change in the future on the TS side of things we can see what we can do in TSLint as well

**Comment by alexeagle** [comment]
I think `private`

**Comment by snolflake** [comment]
@alexeagle this still true?!
Currently in RC5 I'm using a lot of private methods from within the template. Would you advise me to change that?

Thank you...

**Comment by alexeagle** [comment]
Still true. The template is not part of the class, with AoT compilation it is a separate class. So they can't be private.

**Comment by nottinhill** [comment]
This is still not working with Angular Final. We have private template variables. Should we now make them all public just because tslint is too lazy to look into template?

**Comment by alexeagle** [comment]
Having tslint look into Angular templates is complex. To do it right, we really need to use Angular's template expression parser to get 100% compatibility, then hook Angular as an extension in TypeScript language service. We are close to a first version of VSCode support for angular templates. I'd like to figure out the linting story soon after.

**Comment by toepoke** [comment]
For anyone else reading this you can mark the methods as "protected" and the error goes away, but you're not exposing the method outside the class :+1:

**Comment by alexeagle** [comment]
I don't see why `protected` is helpful, you'll still get a type-check error
from the generated .ngfactory.ts file right?

**Comment by toepoke** [comment]
I couldn't say @alexeagle - to be honest I'm using the ionic2 CLI which was giving the same issue.  I would imagine it would solve the same issue .. it's all TypeScript under the hood.

I'm figuring tslint thinks the class may be inherited elsewhere (like a public modifier) and therefore doesn't flag it?

In essence making my methods (that are referenced in the template) no longer error when they're protected :)
